### PR TITLE
fix for issue #250

### PR DESCRIPTION
**Pull request by rwhitmire**

### CODE REVIEW

File: backbone.stickit.js [hunk 1]
```diff
@@ -609,18 +609,31 @@
...
         var removeAllListeners = function() {
           removeCollectionListeners();
           collection.off('stickit:selectRefresh');
+          collection.off('stickit:selectRefreshForView');
           model.off('stickit:selectRefresh');
         };
+
         // Remove previously set event listeners by triggering a custom event
-        collection.trigger('stickit:selectRefresh');
+        function removeCollectionListenersForView(cid) {
```
Comment (akre54) [inline backbone.stickit.js]: No named functions per style

(`var removeCollectionListenersForView = function(cid) {...`)

File: backbone.stickit.js [hunk 2]
```diff
@@ -609,18 +609,31 @@
...
+          if(this.cid == cid){
+            collection.trigger('stickit:selectRefresh');
+          }
+        }
+
+        collection.trigger('stickit:selectRefreshForView', this.cid);
+        collection.once('stickit:selectRefreshForView', removeCollectionListenersForView, this);
```
Comment (akre54) [inline backbone.stickit.js]: Even better, just inline it as an anonymous function here.

File: backbone.stickit.js (hunk 1 above)
Comment (akre54) [inline backbone.stickit.js]: Maybe name the argument `viewCid` just to be explicit *why* it's happening.  A comment here would be great too.

**Comment by akre54** [comment]
Great. Thanks for this. Few minor style tweaks then I'll merge.

**Comment by kgodey** [comment]
@akre54 @rwhitmire I just ran into this bug – any idea when it will be merged?

**Comment by akre54** [comment]
I'm not super familiar with this part of the code, and I'm worried this hackiness will end up being brittle. I'll merge in just to save the headache. Tests like this are always good.

**Comment by rwhitmire** [comment]
The code in that section could definitely use some love. I spent about 2 hours trying to work around various events and this was the most reasonable thing I could come up with unfortunately.

Maybe a better fix would be to use `view.listenTo` so that the event is tied to each view as opposed to a single collection? I have no idea if that would work - just thinking out loud.

**Comment by akre54** [comment]
No worries. If you get some time to clean up the implementation I'm happy to merge in a fix. Thanks for your hard work on this.

**Comment by kgodey** [comment]
Thanks @akre54 and @rwhitmire! Much appreciated.

**Pull request merged**